### PR TITLE
Add route creation and mapping when creating a site with a private bucket

### PR DESCRIPTION
**Pull request by apburnes**
## Description

When a user creates a new site with a private bucket, a new route will be generated and mapped to the `federalist-proxy`, the proxy will point to their private aws bucket endpoint, and will allow users to view the site deployments as normal.

## Implementation

- [x] Add methods to the CF API client to manage routes
- [x] Create and map routes when a site is created
- [x] Remove route when site is deleted

### CODE REVIEW

File: api/controllers/site.js [hunk 1]
```diff
@@ -24,25 +25,26 @@ module.exports = {
     let site;
 
     Promise.resolve(Number(req.params.id)).then((id) => {
-      if (isNaN(id)) {
+      if (_.isNaN(id)) {
```
Comment (amirbey) [inline api/controllers/site.js]: is this necessary? this style isNaN is not used throughout the codebase

File: api/controllers/site.js [hunk 2]
```diff
@@ -24,25 +25,26 @@ module.exports = {
...
-      return authorizer.findOne(req.user, site);
     })
-    .then(() => siteSerializer.serialize(site))
-    .then((siteJSON) => {
-      res.json(siteJSON);
-    })
-    .catch((err) => {
-      res.error(err);
-    });
+      .then((model) => {
```
Comment (amirbey) [inline api/controllers/site.js]: indented 1 too many?

File: api/controllers/site.js [hunk 3]
```diff
@@ -51,7 +53,7 @@ module.exports = {
 
     Promise.resolve(Number(req.params.id))
       .then((id) => {
-        if (isNaN(id)) {
+        if (_.isNaN(id)) {
```
Comment (amirbey) [inline api/controllers/site.js]: s this necessary? this style isNaN is not used throughout the codebase

File: api/controllers/site.js [hunk 4]
```diff
@@ -104,7 +107,7 @@ module.exports = {
     const userId = Number(req.params.user_id);
     let site;
 
-    if (isNaN(siteId) || isNaN(userId)) {
+    if (_.isNaN(siteId) || _.isNaN(userId)) {
```
Comment (amirbey) [inline api/controllers/site.js]: s this necessary? this style isNaN is not used throughout the codebase

File: api/controllers/site.js [hunk 5]
```diff
@@ -124,26 +127,23 @@ module.exports = {
 
       return authorizer.removeUser(req.user, site);
     })
-    .then(() =>
-      SiteMembershipCreator.revokeSiteMembership({ user: req.user, site, userId })
-    )
-    .then(() =>
-      UserActionCreator.addRemoveAction({
+      .then(() => SiteMembershipCreator
```
Comment (amirbey) [inline api/controllers/site.js]: is this a linting issue? 100 characters?

File: api/controllers/site.js [hunk 6]
```diff
@@ -124,26 +127,23 @@ module.exports = {
...
+        .revokeSiteMembership({ user: req.user, site, userId }))
```
Comment (amirbey) [inline api/controllers/site.js]: indented 1 too many?

File: api/controllers/site.js [hunk 7]
```diff
@@ -160,51 +160,52 @@ module.exports = {
     const siteId = Number(req.params.id);
 
     Promise.resolve(siteId).then((id) => {
-      if (isNaN(id)) {
+      if (_.isNaN(id)) {
```
Comment (amirbey) [inline api/controllers/site.js]: s this necessary? this style isNaN is not used throughout the codebase

File: api/controllers/site.js [hunk 8]
```diff
@@ -160,51 +160,52 @@ module.exports = {
...
-      site: siteId,
-      branch: model.defaultBranch,
-    }))
-    .then(() => {
-      if (site.demoBranch) {
-        return Build.create({
-          user: req.user.id,
-          site: siteId,
-          branch: site.demoBranch,
+      .then((model) => {
```
Comment (amirbey) [inline api/controllers/site.js]: indented 1 too many?

File: api/controllers/site.js [hunk 9]
```diff
@@ -160,51 +160,52 @@ module.exports = {
...
+      }))
+      .then(() => {
+        if (site.demoBranch) {
+          return Build.create({
+            user: req.user.id,
+            site: siteId,
+            branch: site.demoBranch,
+          });
+        }
+        return null;
```
Comment (amirbey) [inline api/controllers/site.js]: return null or Promise.resolve() ? or is that the same thing? for consistency, i think the codebase uses Promise.resolve()

File: api/controllers/site.js (hunk 1 above)
Comment (apburnes) [reply api/controllers/site.js]: The `isNaN` was throwing linting error's so I used the underscore version.

File: api/controllers/site.js (hunk 3 above)
Comment (apburnes) [reply api/controllers/site.js]: The `isNaN` was throwing linting error's so I used the underscore version.

File: api/controllers/site.js (hunk 2 above)
Comment (apburnes) [reply api/controllers/site.js]: Lint error thrown on spacing.

File: api/controllers/site.js (hunk 4 above)
Comment (apburnes) [reply api/controllers/site.js]: The isNaN was throwing linting error's so I used the underscore version.

File: api/controllers/site.js (hunk 5 above)
Comment (apburnes) [reply api/controllers/site.js]: Lint error. No line break after arrow function

File: api/controllers/site.js (hunk 6 above)
Comment (apburnes) [reply api/controllers/site.js]: Linting error on spacing

File: api/controllers/site.js (hunk 7 above)
Comment (apburnes) [reply api/controllers/site.js]: The isNaN was throwing linting error's so I used the underscore version

File: api/controllers/site.js (hunk 8 above)
Comment (apburnes) [reply api/controllers/site.js]: Linting again

File: api/controllers/site.js (hunk 9 above)
Comment (apburnes) [reply api/controllers/site.js]: This is linting again... But we can update to `Promise.resolve`

**Pull request merged**